### PR TITLE
Add arange

### PR DESCRIPTION
**Pull request by junjihashimoto**
- Add arange of TensorFactory.
- Add TensorLike instance of Tensor
- Add some dtype-options
- Improve show instance of multi dimensional tensor.

### CODE REVIEW

File: hasktorch/src/Torch/Typed/Functional.hs [hunk 1]
```diff
@@ -4182,7 +4182,10 @@ type family SqueezeAll (shape :: [Nat]) :: [Nat] where
 -- >>> dtype &&& shape $ squeezeAll (ones :: CPUTensor 'D.Float '[2,1,2,1,2])
 -- (Float,[2,2,2])
 -- >>> squeezeAll (ones :: CPUTensor 'D.Float '[2,1,2,1,2])
--- Tensor Float [2,2,2]
+-- Tensor Float [2,2,2] [[[ 1.0000   ,  1.0000   ],
+--                        [ 1.0000   ,  1.0000   ]],
+--                       [[ 1.0000   ,  1.0000   ],
+--                        [ 1.0000   ,  1.0000   ]]]
```
Comment (junjihashimoto) [inline hasktorch/src/Torch/Typed/Functional.hs:4188]: This pr shows multi-dimensional tensor like this doctest.

File: hasktorch/src/Torch/Typed/Functional.hs [hunk 2]
```diff
@@ -4213,12 +4216,6 @@ type SqueezeDim shape dim = SqueezeDimCheck shape dim (SqueezeDimImpl shape dim)
 --
 -- >>> dtype &&& shape $ squeezeDim @1 (ones :: CPUTensor 'D.Float '[2,1,2,1,2])
 -- (Float,[2,2,1,2])
--- >>> squeezeDim @1 (ones :: CPUTensor 'D.Float '[2,1,2,1,2])
--- Tensor Float [2,2,1,2]
--- >>> dtype &&& shape $ squeezeDim @3 (ones :: CPUTensor 'D.Float '[2,1,2,1,2])
```
Comment (junjihashimoto) [inline hasktorch/src/Torch/Typed/Functional.hs:4218]: Should be reverted.

**Comment by tscholak** [comment]
@junjihashimoto you can always request me for review! :)

**Comment by junjihashimoto** [comment]
@tscholak All right. Thx!

**Pull request merged**